### PR TITLE
Issue #66 create Driving_license dao

### LINKED ISSUE #66: Create DrivingLicenseDAO

### PR DESCRIPTION
**Pull request by Nursatb**

### CODE REVIEW

**Review by JustRevolt — APPROVED** [review]

**Review by nikaalin — APPROVED** [review]

File: src/main/java/com/ifmo/epampractice/dao/DrivingLicenseDAO.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Override
+    void update(DrivingLicenseEntity o);
+
+    @Override
+    void delete(DrivingLicenseEntity o);
+
+    @Override
+    void save(DrivingLicenseEntity o);
+}
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/dao/DrivingLicenseDAO.java]: missing newline

File: src/main/java/com/ifmo/epampractice/dao/DrivingLicenseDAO.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+public interface DrivingLicenseDAO extends BasicDaoInterface<DrivingLicenseEntity> {
+    @Override
+    List<DrivingLicenseEntity> getAll();
+
+    @Override
+    DrivingLicenseEntity get(int id);
+
+    @Override
+    void update(DrivingLicenseEntity o);
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/dao/DrivingLicenseDAO.java]: why "o"?

File: src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class DrivingLicenseImpl implements DrivingLicenseDAO {
+
+    private static final String GET_ALL_QUERY = "SELECT * FROM driving_license";
+    private static final String GET_QUERY = "SELECT * FROM driving_license WHERE ID = ?";
+    private static final String UPDATE_QUERY = "UPDATE driving_license SET (Issue_date, Expiration_date, Serial_number) = (?, ?, ?) WHERE ID = ?";
+    private static final String DELETE_QUERY = "DELETE FROM driving_license WHERE ID = ?";
+    private static final String SAVE_QUERY = "INSERT INTO driving_license (Issue_date, Expiration_date, Serial_number) VALUES (?, ?, ?)";
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: too long lines

File: src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+    private static final Logger LOG = LogManager.getLogger(DrivingLicenseImpl.class);
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: Use camelCase for non-constants

Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: Use camelCase for non-constants

Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: Use camelCase for non-constants

File: src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    private DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
+
+    @Override
+    public List<DrivingLicenseEntity> getAll() {
+        List<DrivingLicenseEntity> drivingLicenseEntityArrayList = new ArrayList<>();
+
+        try (Connection connection = dbConnector.getConnection();
+             Statement statement = connection.createStatement(
+                     ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+             ResultSet resultSet = statement.executeQuery(GET_ALL_QUERY)) {
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: Statement and ResultSet should be closed after usage

File: src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+            statement.setInt(1, entity.getId());
+            statement.execute();
+        } catch (SQLException e) {
+            LOG.error(e);
+        }
+    }
+
+    @Override
+    public void save(DrivingLicenseEntity entity) {
+        DBConnectorInterface dbConnector = DBConnectorPostgres.getInstance();
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: you declare dbConnector as class-private field (line 23) and then declare it again here

File: src/main/java/com/ifmo/epampractice/entity/DrivingLicenseEntity.java [hunk 7]
```diff
@@ -1,6 +1,7 @@
 package com.ifmo.epampractice.entity;
 
 import java.time.LocalDate;
+import java.util.Date;
```
Comment (a-deriabin) [inline src/main/java/com/ifmo/epampractice/entity/DrivingLicenseEntity.java]: use LocalDate, don't use Date

**Review by a-deriabin — CHANGES_REQUESTED** [review]

File: src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java (hunk 5 above)
Comment (nikaalin) [reply src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: It'll be auto closed

Comment (a-deriabin) [reply src/main/java/com/ifmo/epampractice/daoimpl/DrivingLicenseImpl.java]: Didn't see all these statements are inside try-with-resources

**Review by a-deriabin — APPROVED** [review]

**Review by PavelEzhkov — APPROVED** [review]

**Pull request merged**